common/PaperSizeProcessor.cc

```diff
@@ -0,0 +1,128 @@
+
+#include <map>
+#include <string>
+#include <tuple>
+
+template <typename T>
+bool Size<T>::operator<(const Size<T>& val) {
+    return std::tie(width, height) < std::tie(val.width, val.height);
+}
+
+Size<int> PaperSizeProcessor::getSizeMM(const std::string& format){
+    auto result = paperSizes.find(format);
+    if(result != paperSizes.end()) {
+        return result->second;
+    } else {
+        return undefinedSize;
+    }
+}
+Size<double> PaperSizeProcessor::getSizeInch(const std::string& format){
+    auto sizeInMM = getSizeMM(format);
+    return {static_cast<double >(sizeInMM.width) / MMtoInch,
+            static_cast<double >(sizeInMM.height) / MMtoInch};
+}
+
+const double PaperSizeProcessor::MMtoInch = 25.4;
+const Size<int> PaperSizeProcessor::undefinedSize = Size<int>(-1, -1);
+
+const std::map<std::string, Size<int>> PaperSizeProcessor::paperSizes = {
+        // A type
+        {"A0", Size<int>(841, 1189)},
+        {"A1", Size<int>(594, 841)},
+        {"A2", Size<int>(420, 594)},
+        {"A3", Size<int>(297, 420)},
+        {"A4", Size<int>(210, 297)},
+        {"A5", Size<int>(148, 210)},
+        {"A6", Size<int>(105, 148)},
+        {"A7", Size<int>(74, 105)},
+        {"A8", Size<int>(52, 74)},
+        {"A9", Size<int>(37, 52)},
+        {"A10", Size<int>(26, 37)},
+
+        // B type
+        {"B0", Size<int>(1000, 1414)},
+        {"B1", Size<int>(707, 1000)},
+        {"B2", Size<int>(500, 707)},
+        {"B3", Size<int>(353, 500)},
+        {"B4", Size<int>(250, 353)},
+        {"B5", Size<int>(176, 250)},
+        {"B6", Size<int>(125, 176)},
+        {"B7", Size<int>(88, 125)},
+        {"B8", Size<int>(62, 88)},
+        {"B9", Size<int>(44, 62)},
+        {"B10", Size<int>(31, 44)},
+
+        // C type
+        {"C0", Size<int>(917, 1297)},
+        {"C1", Size<int>(648, 917)},
+        {"C2", Size<int>(458, 648)},
+        {"C3", Size<int>(324, 458)},
+        {"C4", Size<int>(229, 324)},
+        {"C5", Size<int>(162, 229)},
+        {"C6", Size<int>(114, 162)},
+        {"C7", Size<int>(81, 114)},
+        {"C8", Size<int>(57, 81)},
+        {"C9", Size<int>(40, 57)},
+        {"C10", Size<int>(28, 40)},
+
+        // D type. (German extension)
+        {"D0", Size<int>(771, 1090)},
+        {"D1", Size<int>(545, 771)},
+        {"D2", Size<int>(385, 545)},
+        {"D3", Size<int>(272, 385)},
+        {"D4", Size<int>(192, 272)},
+        {"D5", Size<int>(136, 192)},
+        {"D6", Size<int>(96, 136)},
+        {"D7", Size<int>(68, 96)},
+        {"D8", Size<int>(48, 68)},
+
+        // North American paper sizes
+        {"Letter", Size<int>(216, 279)},
+        {"Legal", Size<int>(216, 356)},
+        {"Tabloid", Size<int>(279, 432)},
+        {"Ledger", Size<int>(432, 279)},
+        {"Junior Legal", Size<int>(127, 203)},
+        {"Half Letter/Memo", Size<int>(140, 216)},
+        {"Government Letter", Size<int>(203, 267)},
+        {"Government Legal", Size<int>(216, 330)},
```

Comment:
These are not localizable in this form. Though I don't know whether they should be localizable, since these are probably actual paper names that only exists in the respective language.

Reply:
As fasr as i know we shouldn't localize it